src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+
+    static final String KAFKA_INSTANCE_NAME = "cl-e2e-" + Environment.LAUNCH_KEY;
+    private OpenShiftClient oc;
+
+    private static int maxStandardInstancesInCluster;
+    private static int maxDeveloperInstancesInCluster;
+    private static final String KAFKAS_MGMT_21_CODE = "KAFKAS-MGMT-21";
+    private static final String PLAN_STANDARD = "standard.x1";
+    private static final String INSTANCE_NAME_PREFIX = "cl-e2e-placeholder-";
+    private KafkaMgmtApi kafkaMgmtApi;
+
+    @BeforeClass
+    @SneakyThrows
+    public void bootstrap() {
+
+        log.info("build config");
+        Config config = new ConfigBuilder()
+            .withMasterUrl("https://api.mk-stage-0622.bd59.p1.openshiftapps.com:6443")
+            .withOauthToken(Environment.PROMETHEUS_WEB_CLIENT_ACCESS_TOKEN)
+            .withTrustCerts(true)
+            .build();
+
+        log.info("init openshift client");
+        oc = new KubernetesClientBuilder().withConfig(config).build().adapt(OpenShiftClient.class);
+
+        log.info("init cluster capacity info");
+        maxStandardInstancesInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.standard);
+        log.debug("Max standard instances according to MKAgent capacity '{}'", maxStandardInstancesInCluster);
+        maxDeveloperInstancesInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.standard);
+        log.debug("Max developer instances according to MKAgent capacity '{}'", maxDeveloperInstancesInCluster);
+
+        log.info("init kafka management");
+        var apps = ApplicationServicesApi.applicationServicesApi(
+            Environment.PRIMARY_USERNAME,
+            Environment.PRIMARY_PASSWORD);
+        kafkaMgmtApi = apps.kafkaMgmt();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown() {
+
+        // delete kafka instance
+        try {
+            KafkaMgmtApiUtils.cleanKafkaInstance(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+        } catch (Throwable t) {
+            log.error("clean main kafka instance error: ", t);
+        }
+    }
+
+    @DataProvider(name = "managedKafkaTypes")
+    public Object[][] managedKafkaTypeDataProvider() {
+        return new Object[][]{
+            {"standard"},
+            {"developer"}};
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    @SneakyThrows
+    public void testReservedDeploymentExistence(String mkType) {
+
+        String namespace = String.format("reserved-kafka-%s-1", mkType);
+        log.info("evaluating namespace '{}'", namespace);
+
+        String managedKafkaName = String.format("reserved-kafka-%s-1", mkType);
+        log.info("managed kafka reserved deployment '{}'", managedKafkaName);
+        Optional<ManagedKafka> mkOptional = FleetshardUtils.managedKafka(oc).inNamespace(namespace).list().getItems().stream().filter(ManagedKafka::isReserveDeployment).findAny();
+        Assert.assertTrue(mkOptional.isPresent(), "reserved deployment is not present");
```

Comment:
We might need to extend this test to actually check that:
- for `standard` reserved instance (there are 3 kafka brokers pods & 3 zookeepers pods)
- for `developer` reserved instance (there is 1 Kafka broker pod & 1 zookeeper pod)

Reply:
that is no problem, (we are capable of listing pods in all namespaces already so i can add this check).

Reply:
Thanks.